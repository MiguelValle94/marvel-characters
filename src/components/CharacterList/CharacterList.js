import React from 'react';
import PropTypes from 'prop-types';
import CharacterListItem from './CharacterListItem/CharacterListItem';

export default function CharacterList({ characters }) {
  return (
    <div>
      <ul>
        {characters.map((character) => (
          <CharacterListItem character={character} key={character.id} />
        ))}
      </ul>
    </div>
  );
}

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object).isRequired
};
