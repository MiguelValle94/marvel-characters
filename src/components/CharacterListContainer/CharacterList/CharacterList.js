import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import CharacterListItem from './CharacterListItem';

export default function CharacterList({ characters }) {
  return (
    <div>
      <ul>
        {characters.map((character) => (
          <Link to={`/${character.id}`} key={`link-${character.id}`}>
            <CharacterListItem character={character} key={character.id} />
          </Link>
        ))}
      </ul>
    </div>
  );
}

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.object).isRequired
};
