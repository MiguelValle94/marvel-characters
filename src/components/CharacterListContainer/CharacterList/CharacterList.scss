.character-list {
  display: grid;
  // grid-template: 1fr / 1;
  list-style: none;
  margin: 0;
  padding: 0;

  &__item {
    padding: 10px;
  }

  @media only screen and (max-width: 767px) and (orientation: landscape),
    (min-width: 768px) {
    grid-template: 1fr / repeat(2, 50%);
  }

  @media only screen and (min-width: 992px) {
    grid-template: 1fr / repeat(4, 25%);
  }

  @media only screen and (min-width: 1200px) {
    grid-template: 1fr / repeat(5, 20%);
  }
}
