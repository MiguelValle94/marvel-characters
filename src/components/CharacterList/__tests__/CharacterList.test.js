import React from 'react';
import { shallow } from 'enzyme';

import CharacterList from '../CharacterList';

const setUpShallow = (characters) => {
  const component = shallow(<CharacterList characters={characters} />);
  return component;
};

describe('CharacterList', () => {
  let component;

  it('should display characters list', () => {
    component = setUpShallow([
      { id: 1, name: 'Ironman' },
      { id: 2, name: 'Thor' },
      { id: 3, name: 'Black Widow' }
    ]);
    expect(component).toMatchSnapshot();
  });
});
