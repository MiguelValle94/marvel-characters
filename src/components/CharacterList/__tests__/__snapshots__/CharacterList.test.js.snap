// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList should display characters list 1`] = `
<div>
  <ul>
    <CharacterListItem
      character={
        Object {
          "id": 1,
          "name": "Ironman",
        }
      }
      key="1"
    />
    <CharacterListItem
      character={
        Object {
          "id": 2,
          "name": "Thor",
        }
      }
      key="2"
    />
    <CharacterListItem
      character={
        Object {
          "id": 3,
          "name": "Black Widow",
        }
      }
      key="3"
    />
  </ul>
</div>
`;
